ios/objc/src/TorchIValue.mm

```diff
@@ -0,0 +1,191 @@
+#include <vector>
+#import "TorchIValue.h"
+#import "TorchIValue+Internal.h"
+#import "TorchTensor.h"
+#import "TorchTensor+Internal.h"
+#import <Pytorch/Pytorch.h>
+
+#define IVALUE_TYPE(_) \
+    _(None) \
+    _(Bool) \
+    _(Int) \
+    _(Double) \
+    _(Tensor) \
+    _(BoolList) \
+    _(IntList) \
+    _(DoubleList)\
+    _(TensorList)
+
+#define IVALUE_SCALAR_TYPE(_) \
+    _(Bool, bool, bool) \
+    _(Int, int, int64_t) \
+    _(Double, double,double) \
+
+@implementation TorchIValue {
+    std::shared_ptr<at::IValue> _impl;
+}
+
++ (instancetype)newWithNone{
+    auto impl = std::make_shared<at::IValue>();
+    if(!impl) {
+        return nil;
```

Comment:
If we support `nil` return values, the return type must be `nullable instancetype`